libs/aws/langchain_aws/utils.py

```diff
@@ -1,21 +1,39 @@
 import re
 from typing import Any, List
 
+from packaging import version
+
 
 def enforce_stop_tokens(text: str, stop: List[str]) -> str:
     """Cut off the text as soon as any stop words occur."""
     return re.split("|".join(stop), text, maxsplit=1)[0]
 
 
-def _get_anthropic_client() -> Any:
+def anthropic_tokens_supported() -> bool:
+    """Check if we have all requirements for Anthropic count_tokens() and get_tokenizer()."""
     try:
         import anthropic
     except ImportError:
-        raise ImportError(
-            "Could not import anthropic python package. "
-            "This is needed in order to accurately tokenize the text "
-            "for anthropic models. Please install it with `pip install anthropic`."
-        )
+        return False
+
+    anthropic_version = version.parse(anthropic.__version__)
+    if anthropic_version > version.parse("0.38.0"):
+        return False
+    else:
+        httpx_import_msg = "httpx<=0.27.2 is required."
+        try:
+            import httpx
+        except ImportError:
+            raise ImportError(httpx_import_msg)
+        httpx_version = version.parse(httpx.__version__)
+        if httpx_version > version.parse("0.27.2"):
+            raise ImportError(httpx_import_msg)
+        else:
+            return True
```

Comment:
Can simplify this by getting rid of the 2 else as we are either raising or return early.
```suggestion
    if version.parse(anthropic.__version__) > version.parse("0.38.0"):
        return False

    try:
        import httpx
        if httpx_version > version.parse("0.27.2"):
            raise ImportError()
    except ImportError:
        raise ImportError("httpx<=0.27.2 is required.")

    return True
```